File: Problem/Problem/66-plus-one/plus-one.cpp
class Solution {
public:
    vector<int> plusOne(vector<int>& digits)
    {
        vector <int> d;
        int flag = 1;
        int i = digits.size() - 1;
        while (i >= 0)
        {
            if (flag == 0)
                break;
            digits[i]++;
            if (digits[i] == 10)
            {
                digits[i] = 0;
                flag = 1;
            }
            else
                flag = 0;
            i--;
        }
        if (flag)
            d.push_back(1);
         i = 0;
        while (i < digits.size())
            {
                d.push_back(digits[i]);
                i++;
            }
        return (d);
    }
};

File: 66-plus-one/plus-one.py
class Solution(object):
    def plusOne(self, digits):
        """
        :type digits: List[int]
        :rtype: List[int]
        """
        rest = 0
        for i in range(len(digits)-1, -1, -1):
            digits[i] += 1
            if digits[i] >= 10:
                digits[i] = 0
                rest = 1
            else :
                rest = 0
                break
        if rest :
            digits.insert(0, 1)
        print(digits)
        return digits